### PR TITLE
Add missing changelog entries

### PR DESCRIPTION
**Pull request by fredpi**
Usually, these changelog entries should have been added with their respective PRs, but as this hasn't been the case over the last year, I have now added all missing changelog entries.

If I missed something here or if I should change something, please let me know.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,15 +8,22 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/) a
...
 
-- None
+- Add support for SF Symbols 2.2, 3.0, 3.1 & 3.2 (By [Steven Magdy](https://github.com/StevenMagdy))
+- Add support for explicit symbol localization (By [David Knothe](https://github.com/knothed))
+- Add documentation for available layersets & improve documentation in general. (By [Frederick Pietschmann](https://github.com/fredpi))
 
 ### Changed
 
-- None
+- Deprecate `allCases: [SFSymbol]` in favor of new `allSymbols: Set<SFSymbol>`. (By [Steven Magdy](https://github.com/StevenMagdy))
```
Comment (knothed) [inline CHANGELOG.md:17]: `allSymbols` is actually of type `Set<LocalizableSymbol>` after #98.

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,15 +8,22 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/) a
...
+- Transfer repository to new location: https://github.com/SFSafeSymbols/SFSafeSymbols & simplify contributions to this repository. (By [Frederick Pietschmann](https://github.com/fredpi))
+- Change the type of `SFSymbol` from `enum` to `struct`, using `static let ...` instead of `case` for individual symbols. This allows for specification of custom symbols as `static let`s in `extension`s to `SFSymbol` and fixes an issue that prevented `rawValue` initialization of symbols. (By [Steven Magdy](https://github.com/StevenMagdy))
```
Comment (knothed) [inline CHANGELOG.md]: After #98, `SFSymbol` is a `class`.

**Review by knothed — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (knothed) [reply CHANGELOG.md:17]: Not anymore.

**Comment by fredpi** [comment]
I will release once this is merged. Are you okay with this PR @knothed, @StevenMagdy?

And another question: I think, the new release should be called `3.2` to stay in sync with Apple's versioning. Are you okay with that?

**Review by knothed — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -8,15 +8,23 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/) a
...
 
 ### Changed
 
-- None
+- Deprecate `allCases: [SFSymbol]` in favor of new `allSymbols: Set<SFSymbol>`. (By [Steven Magdy](https://github.com/StevenMagdy))
+- Transfer repository to new location: https://github.com/SFSafeSymbols/SFSafeSymbols & simplify contributions to this repository. (By [Frederick Pietschmann](https://github.com/fredpi))
+- Change the type of `SFSymbol` from `enum` to `class`, using `static let ...` instead of `case` for individual symbols. This allows for specification of custom symbols as `static let`s in `extension`s to `SFSymbol` and fixes an issue that prevented `rawValue` initialization of symbols. (By [Steven Magdy](https://github.com/StevenMagdy))
+- Improve internal code generation to allow for quick adjustments to new SF Symbols versions. (By [ddddxxx](https://github.com/ddddxxx))
+- Drop dedicated `SFSafeSymbols-Dynamic` product in the `Package.swift` in favour of a new unified `SFSafeSymbols` product with automatic selection between static / dynamic linking. (By [Steven Magdy](https://github.com/StevenMagdy))
+- Raise minimum platform requirements to 2017 platforms. (By [Steven Magdy](https://github.com/StevenMagdy))
```
Comment (StevenSorial) [inline CHANGELOG.md]: The result of that PR was just **lowering** watchOS from 6.0 to 4.0

**Comment by StevenSorial** [comment]
LGTM. just added one comment for OS availability.
I have one more PR refactoring `allCases` that I want to make in an attempt to reduce the package size. Hopefully, I can submit it this weekend,

File: CHANGELOG.md (hunk 3 above)
Comment (fredpi) [reply CHANGELOG.md]: Oops, I should have taken a closer look 😬

**Review by StevenSorial — APPROVED** [review]

**Pull request merged**